File: src/main/java/no/nav/bidrag/beregn/samvaersfradrag/beregning/SamvaersfradragBeregning.java
package no.nav.bidrag.beregn.samvaersfradrag.beregning;

import no.nav.bidrag.beregn.samvaersfradrag.bo.GrunnlagBeregning;
import no.nav.bidrag.beregn.samvaersfradrag.bo.ResultatBeregning;

public interface SamvaersfradragBeregning {

  ResultatBeregning beregn(GrunnlagBeregning grunnlagBeregning);

  static SamvaersfradragBeregning getInstance(){
    return new SamvaersfradragBeregningImpl();
  }
}

File: src/main/kotlin/no/nav/bidrag/beregn/samvaersfradrag/beregning/SamvaersfradragBeregning.kt
package no.nav.bidrag.beregn.samvaersfradrag.beregning

import no.nav.bidrag.beregn.samvaersfradrag.bo.GrunnlagBeregning
import no.nav.bidrag.beregn.samvaersfradrag.bo.ResultatBeregning

fun interface SamvaersfradragBeregning {
    fun beregn(grunnlag: GrunnlagBeregning): ResultatBeregning

    companion object {
        fun getInstance(): SamvaersfradragBeregning {
            return SamvaersfradragBeregningImpl()
        }
    }
}

